pipeline/types/engine.py

```diff
@@ -48,3 +48,10 @@ class MatchingOutput:
     release: ReleaseTag
     users: List[User]
     matches: List[Match]
+
+
+@dataclass
+class MatchingInternalAnalytics:  # More metrics will be added in the future
+    n_proposed_matches_per_user: List[
+        Tuple[User, int]
+    ]  # The list will be a tuple containing the user and their number of proposed matches
```

Comment:
You can also set a default value for fields in dataclasses.

For a list:

```python
name: Type = field(default_factory=list)
```

For a dictionary:

```python
name: Type = field(default_factory=dict)
```

Reply:
Yep. I agree. I already revised the dataclass to be a Dict instead of a List. I'll set the default value to be a dict as well.